src/rgw/rgw_op.cc

```diff
@@ -1029,6 +1029,10 @@ int handle_cloudtier_obj(req_state* s, const DoutPrefixProvider *dpp, rgw::sal::
       uint64_t epoch = 0;
       op_ret = get_system_versioning_params(s, &epoch, NULL);
       ldpp_dout(dpp, 20) << "getting versioning params tier placement handle cloud tier" << op_ret << dendl;
+      if (op_ret < 0) {
+        s->err.message = "failed to get versioning params";
```

Comment:
maybe this could be in log message and in s->err.message could be "failed to restore object".